pkg/middlewares/aws_default_network_acl_rule.go

```diff
@@ -0,0 +1,81 @@
+package middlewares
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	"github.com/sirupsen/logrus"
+)
+
+// Default network acl rules should not be shown as unmanaged as they are present by default
+// This middleware ignores default network acl rules from unmanaged resources if they are not managed by IaC
+type AwsDefaultNetworkACLRule struct{}
+
+func NewAwsDefaultNetworkACLRule() AwsDefaultNetworkACLRule {
+	return AwsDefaultNetworkACLRule{}
+}
+
+func (m AwsDefaultNetworkACLRule) Execute(remoteResources, resourcesFromState *[]*resource.Resource) error {
+
+	newRemoteResources := make([]*resource.Resource, 0)
+
+	for _, remoteResource := range *remoteResources {
+		// Ignore all resources other than ACL rules
+		if remoteResource.ResourceType() != aws.AwsNetworkACLRuleResourceType {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		// Ignore non default ACL rules
+		if !m.isDefaultACLRule(remoteResource) {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		// Check if resource is managed by IaC
+		existInState := false
+		for _, stateResource := range *resourcesFromState {
+			if remoteResource.Equal(stateResource) {
+				existInState = true
+				break
+			}
+		}
+
+		// Include resource if it's managed in IaC
+		if existInState {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		// Else, resource is not added to newRemoteResources slice so it will be ignored
+		logrus.WithFields(logrus.Fields{
+			"id":             remoteResource.ResourceId(),
+			"type":           remoteResource.ResourceType(),
+			"network_acl_id": *remoteResource.Attrs.GetString("network_acl_id"),
+		}).Debug("Ignoring default ACL rule as it is not managed by IaC")
+	}
+
+	*remoteResources = newRemoteResources
+
+	return nil
+}
+
+func (m *AwsDefaultNetworkACLRule) isDefaultACLRule(res *resource.Resource) bool {
+
+	if number := res.Attrs.GetFloat64("rule_number"); number != nil && int(*number) != 32767 {
+		return false
+	}
+
+	if action := res.Attrs.GetString("rule_action"); action != nil && *action != "deny" {
+		return false
+	}
+
+	if cidr := res.Attrs.GetString("cidr_block"); cidr != nil && *cidr != "0.0.0.0/0" {
+		return false
+	}
+
+	if proto := res.Attrs.GetString("protocol"); proto != nil && *proto != "-1" {
+		return false
+	}
+
```

Comment:
It misses a case with ipv6_cidr_block. Test with this tf file:

```
resource "aws_vpc" "foo" {
    cidr_block = "10.1.0.0/16"
    assign_generated_ipv6_cidr_block = true
}

resource "aws_network_acl" "bar" {
  vpc_id = aws_vpc.foo.id
}

resource "aws_network_acl_rule" "bar" {
  network_acl_id = aws_network_acl.bar.id
  rule_number    = 200
  egress         = false
  protocol       = "tcp"
  rule_action    = "allow"
  # cidr_block     = aws_vpc.foo.cidr_block
  ipv6_cidr_block = aws_vpc.foo.ipv6_cidr_block
  from_port      = 22
  to_port        = 22
}
```

Reply:
I don't think this is relevant here, actually there is no default IPv6 rules

Reply:
Are you talking about the default one on each region or the default one created once you create a new VPC ?

Reply:
I added ipv6 handling here you are right